Clarify background rate per livetime or obstime

cdeil opened the pull request:
The specs for background rate aren't clear if it's per livetime or per observation time:
http://gamma-astro-data-formats.readthedocs.io/en/latest/irfs/full_enclosure/bkg/index.html

This needs to be clear for science tools to correctly compute predicted background counts.

@jknodlseder - How do you interpret the values in ctools?

@gernotmaier - How do you fill background in the CTA IRF FITS exporters?

cdeil commented:
I've added one clarifying sentence here that background rate is per observation time, not livetime in ecf93f5 . I've put it in a "notes" section at the end, like we already have on other pages, such as e.g. the EVENTS page.

@TarekHC - Please review. Let me know if there's something that should be improved, or merge if it looks good to you.

@lmohrmann - You're working on HESS background rate models - you're filling the rates per observation time (not livetime), right?

@jknodlseder - I think ctools already uses background rate like this, so nothing to do there, no?

(I think all code in Gammapy to produce and use background rate models, like [here](http://docs.gammapy.org/en/latest/api/gammapy.cube.make_background_cube.html) added today, is already using it as a rate per obs time, so nothing left to adjust there)

TarekHC commented on source/irfs/full_enclosure/bkg/index.rst:
```diff
@@ -91,3 +83,14 @@ declare the type of HDU:
+It is already a count rate, it doesn't need to be multiplied with
+effective area to obtain predicted counts, like gamma-ray flux and
+surface brightness models do.
+The rate is computed per observation time (without any dead time correction,
```
dead-time correction

TarekHC commented:
Everything looks ok to me. Merged.

jknodlseder commented:
ctools and GammaLib interpret background rate per livetime

> Le 23 oct. 2017 à 21:48, Christoph Deil <notifications@github.com> a écrit :
> 
> I've added one clarifying sentence here that background rate is per observation time, not livetime in ecf93f5 <https://github.com/open-gamma-ray-astro/gamma-astro-data-formats/commit/ecf93f547922acc0f80e7f3d5f87870ebd8fa9cf> . I've put it in a "notes" section at the end, like we already have on other pages, such as e.g. the EVENTS page.
> 
> @TarekHC <https://github.com/tarekhc> - Please review. Let me know if there's something that should be improved, or merge if it looks good to you.
> 
> @lmohrmann <https://github.com/lmohrmann> - You're working on HESS background rate models - you're filling the rates per observation time (not livetime), right?
> 
> @jknodlseder <https://github.com/jknodlseder> - I think ctools already uses background rate like this, so nothing to do there, no?
> 
> (I think all code in Gammapy to produce and use background rate models, like here <http://docs.gammapy.org/en/latest/api/gammapy.cube.make_background_cube.html> added today, is already using it as a rate per obs time, so nothing left to adjust there)
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/open-gamma-ray-astro/gamma-astro-data-formats/pull/97#issuecomment-338775639>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AC2oV4COdUNo9iHGBQVCQENdaO_HFo8sks5svO2KgaJpZM4QCl2O>.
>

jknodlseder commented:
GammaLib / ctools has the convention that counts = rate * livetime, hence all rates are per livetime.

> Le 23 oct. 2017 à 12:05, Christoph Deil <notifications@github.com> a écrit :
> 
> The specs for background rate aren't clear if it's per livetime or per observation time:
> http://gamma-astro-data-formats.readthedocs.io/en/latest/irfs/full_enclosure/bkg/index.html <http://gamma-astro-data-formats.readthedocs.io/en/latest/irfs/full_enclosure/bkg/index.html>
> This needs to be clear for science tools to correctly compute predicted background counts.
> 
> @jknodlseder <https://github.com/jknodlseder> - How do you interpret the values in ctools?
> 
> @GernotMaier <https://github.com/gernotmaier> - How do you fill background in the CTA IRF FITS exporters?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/open-gamma-ray-astro/gamma-astro-data-formats/issues/97>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AC2oV6shik6dxB6lR5r7zgqP2LHT0KNfks5svGTcgaJpZM4QCl2O>.
>

cdeil commented:
@GernotMaier - You could consider adding one sentence "background rate is given per observation time, not livetime" to the CTA IRF description document you wrote, to make it clear what is given.

@jknodlseder - Would you consider changing Gammalib / ctools to have background rate per observation time instead of per livetime?

---

Given that CTA dead time fraction will be very small (~1%) and at the moment is not considered at all in the simulations, it doesn't really matter for results (and in addition most analyses have a free scale parameter on the background, i.e. results don't pend on the absolute rate in the background IRF model). So making any changes here isn't high priority, but someone will have to pin down the IACT DL3 details at some point. In the end it's just a matter of taste where the book-keeping for deadtime fraction is done, and for now we have decided to just follow in the spec what DL3 producers already do (i.e. background rate per observation time). @TarekHC already linked to #62 above where deadtime fraction is discussed in the context of gammas / effective area. I'm not sure, but again I think actually what we have in the open spec is exactly what DL3 producers already do, and might be sufficient for CTA (if they want to absorb dead time fraction in effective area, they can just set DEADC header key to 0).

GernotMaier commented:
I've updated the document - there are a few more minor things to do before sending out a new version. For those with access, the most recent version can always be found on the CTA svn:

https://forge.in2p3.fr/projects/cta_analysis-and-simulations/repository/changes/DOC/InternalReports/IRFReports/trunk/cta-aswg-IRFreport.pdf

jknodlseder commented:
I would not like to change the definition of counts and rate as this would be inconsistent between source and background counts.

> Le 29 oct. 2017 à 14:41, Christoph Deil <notifications@github.com> a écrit :
> 
> @GernotMaier <https://github.com/gernotmaier> - You could consider adding one sentence "background rate is given per observation time, not livetime" to the CTA IRF description document you wrote, to make it clear what is given.
> 
> @jknodlseder <https://github.com/jknodlseder> - Would you consider changing Gammalib / ctools to have background rate per observation time instead of per livetime?
> 
> Given that CTA dead time fraction will be very small (~1%) and at the moment is not considered at all in the simulations, it doesn't really matter for results (and in addition most analyses have a free scale parameter on the background, i.e. results don't pend on the absolute rate in the background IRF model). So making any changes here isn't high priority, but someone will have to pin down the IACT DL3 details at some point. In the end it's just a matter of taste where the book-keeping for deadtime fraction is done, and for now we have decided to just follow in the spec what DL3 producers already do (i.e. background rate per observation time). @TarekHC <https://github.com/tarekhc> already linked to #62 <https://github.com/open-gamma-ray-astro/gamma-astro-data-formats/issues/62> above where deadtime fraction is discussed in the context of gammas / effective area. I'm not sure, but again I think actually what we have in the open spec is exactly what DL3 producers already do, and might be sufficient for CTA (if they want to absorb dead time fraction in effective area, they can just set DEADC header key to 0).
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/open-gamma-ray-astro/gamma-astro-data-formats/pull/97#issuecomment-340263151>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AC2oV7SYCQKSvPqDVq1Meha2kccSvEnCks5sxIBtgaJpZM4QCl2O>.
>

cdeil commented:
I hadn't realised that in Gammapy we had introduced a background computation as `counts = rate * livetime`. Following the discussion above, we have changed to `counts = rate * obstime` in https://github.com/gammapy/gammapy/pull/1861 .

Overall by now I like the proposal some people made in #62 to get rid of DEADC at the DL3 level, and just define AEFF and BKG rate such that it contains dead-time effects, and science tools only need to use obstime. Especially if we use GTI tables, it's simpler if one doesn't also need to query the EVENTS header to get a DEADC key to get a livetime and compute and exposure or predicted background.

@jknodlseder - For ctools are you willing to change to this definition? Or do you prefer to keep as-is, and wait for an official CTA decision on this before changing the software?
